### DISCUSSION TITLE
User experience upon registration/login

### DISCUSSION THREAD

**Discussion by timea-solid** [solid/solidos#54] [Archive] [answered]
I noticed an interesting user experience on the registration/login and I also noticed some mentions about pages from the server and from SolidOS. To try to understand better what is going on I created a diagram: https://drive.google.com/file/d/1DlKyTinKbvZaFB6qu5KdbSfG4ggB8HeQ/view?usp=sharing
![Solid onboarding drawio(1)](https://user-images.githubusercontent.com/4144203/132177619-31356b53-b413-4252-b41b-6a0d26f650b5.png)

I have some questions about this:
- did I get the diagram right?
- which pages are served from the server (NSS in this case) and which from SolidOS Databrowser?
- is there a reason why one needs to sign in twice if coming from registration? Is it because of the ID provider?
- which is the 1st page the user should see? (there seems to be 2).
- how do I get to the data folder list view when I am logged in?
    P.s. Will it all change with the new auth?

**Comment by timbl** [comment]
(The diagrams for some reason are very big but very low resolution in Chrome, but downloaded, the PNG is legible on a laptop.)
I am disinclined to go into this flow in detail, as the flow may have changed in the upgraded auth version.

**Comment by timea-solid** [comment, marked as answer]
After the Auth upgrade work the user flow looks like this:

![SolidOS onboarding(1)](https://user-images.githubusercontent.com/4144203/140076849-85f76fb0-43cb-4d81-bc8c-354bf756a962.jpg)
